compiler/src/dmd/location.d

```diff
@@ -0,0 +1,136 @@
+/**
+ * Encapsulates file/line/column locations.
+ *
+ * Copyright:   Copyright (C) 1999-2022 by The D Language Foundation, All Rights Reserved
+ * Authors:     $(LINK2 https://www.digitalmars.com, Walter Bright)
+ * License:     $(LINK2 https://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Source:      $(LINK2 https://github.com/dlang/dmd/blob/master/src/dmd/globals.d, _location.d)
+ * Documentation:  https://dlang.org/phobos/dmd_location.html
+ * Coverage:    https://codecov.io/gh/dlang/dmd/src/master/src/dmd/location.d
+ */
+
+module dmd.location;
+
+import dmd.common.outbuffer;
+import dmd.root.filename;
+import dmd.globals;
+
+version (DMDLIB)
+{
+    version = LocOffset;
+}
+
+/**
+A source code location
+
+Used for error messages, `__FILE__` and `__LINE__` tokens, `__traits(getLocation, XXX)`,
+debug info etc.
+*/
+struct Loc
+{
+    /// zero-terminated filename string, either absolute or relative to cwd
+    const(char)* filename;
+    uint linnum; /// line number, starting from 1
+    uint charnum; /// utf8 code unit index relative to start of line, starting from 1
+    version (LocOffset)
+        uint fileOffset; /// utf8 code unit index relative to start of file, starting from 0
+
+    static immutable Loc initial; /// use for default initialization of const ref Loc's
+
+nothrow:
```

Comment:
Lots of methods have nothrow on them, as well as this at global scope, there could be some simplification here!

Reply:
I know, but pls save it for a later PR.